pages/college_predictor.js

```diff
@@ -5,26 +5,35 @@ import getConstants from "../constants";
 
 const CollegePredictor = () => {
     const router = useRouter();
-    console.log(router.query);
-    const { rank, category, roundNumber, gender, exam, stateName } = router.query;
+    // console.log(router.query);
+    const { rank, category, roundNumber, exam, gender = "", stateName = "" } = router.query;
     const [filteredData, setFilteredData] = useState([]);
     const [isLoading, setIsLoading] = useState(true);
 
     useEffect(() => {
         const fetchData = async () => {
             try {
-                const response = await fetch("/data/" + category + ".json"); // Replace with the path to your category JSON file
+                let exam_fol;
+                if (exam === "JEE Main" || exam === "JEE Advanced") {
+                    exam_fol = "JEE"
+                } else exam_fol = "NEET";
+                const response = await fetch("/data/" + exam_fol + "/" + category + ".json"); // Replace with the path to your category JSON file
                 const data = await response.json();
 
                 // Filter the data based on round number
                 const dataForGivenQuery = data.filter((item) => {
-                    const itemRound = item["Round"];
-                    const itemGender = item["Gender"];
-                    const itemState = item["State"];
+                    const itemRound = parseInt(item["Round"], 10);
                     const itemExam = item["Exam"];
-                    const itemQuota = item["Quota"];
-                    const checkForState = (itemState == stateName) || (stateName == "All India") || (itemQuota == "OS") || (itemQuota == "AI");
-                    return itemRound == roundNumber && itemGender == gender && itemExam == exam && checkForState;
+                    if (exam != "NEET") {
```

Comment:
Replace with checking if exam is main or advanced. Clearer for readers and as we add more exams it will make sense